### ISSUE TITLE
updateAll AssertionError: The where argument must be an object

### ISSUE THREAD

**Issue by rodplata** [mean-expert-official/loopback-sdk-builder#356]
#### What type of issue are you creating?
- [x] Bug
- [ ] Enhancement
- [ ] Question

#### What version of this module are you using?
- [ ] 2.0.10 (Stable)
- [x] 2.1.0-rc.n (2.1 Release Candidate n)
- [ ] Other

Write other if any:
 package.js
```
"loopback": "^3.2.1",
"loopback-boot": "^2.6.5",
"loopback-component-explorer": "^2.4.0",
"loopback-connector-cloudant": "^1.0.11",
```
#### Please add a description for your issue:

After run the new version of the SDK builder all the updateAll function fails in the API with the error:
`AssertionError: The where argument must be an object`

**Comment by rodplata** [comment]
I discovered that this error is because of the change in the request function
```
// Separate where object from url params and add to search query
if (urlParams.where) {
  headers.append('where', JSON.stringify(urlParams.where));
  delete urlParams.where;
}
```

**Comment by dmastag** [comment]
So, how should the request be made then ?

**Comment by jonathan-casarrubias** [comment]
Yes that was a pull request that broke several use cases,

Since there are multiple issues related to this patch I will remove it, but I can not do it right now, since the the projects that actually pays needs to be done, once I finish I will come and take a couple of days to kill all of these issues.

**Comment by jonathan-casarrubias** [comment]
I'm adding this as bug since I don't feel where should be on the header

**Comment by Destreyf** [comment]
@jonathan-casarrubias Just to give some feedback, at the time that the where statement was added for the counts request (should only be there) if i had a complex where statement loopback was actually not reading it properly and my count query would fail.  Adding it as a header actually fixes that behavior.

I can test it on newer versions of loopback as we've migrated to 3.X and i am in the process of working out all the kinks from the update.
